#87: Expand specification examples

brendanreardon opened the pull request:
This is a Pull Request for [Issue 87: Add examples for Cat-VRS data classes](https://github.com/ga4gh/cat-vrs/issues/87). It also adds Pull Request [143: #133: Import examples into readthedocs](https://github.com/ga4gh/cat-vrs/pull/143) into 1.0.0-ballot.2025-03.

Here, we revise current examples to have all optional properties of underlying VRS SequenceLocation and SequenceReference objects populated, write new examples, and incorporate them into the readthedocs documentation. 

In all cases, SequenceLocation objects now have `id`, `description`, and `extensions` populated, where `extensions` will list 0- and 1-based representations of the location, and SequenceReference objects will have `id`, `name`, `description`, `aliases`, and `residueAlphabet` populated. This was done in response to [Issue 144: Not clear how to extract precise location information from a location.id](https://github.com/ga4gh/cat-vrs/issues/144). 

New examples include:
- categoricalCnv-ex4, TP53 Loss
- braf-V600 
Both of these examples have corresponding "annotated" examples with inline comments that describe how each property was populated.

Examples shown for Constraints _only_ show the constraint within a given example. Inline examples highlighted are:
- DefiningAlleleConstraint: NM_004958.4(MTOR):c.5992_5993del (p.Met1998fs), NC_000001.11:g.1699974C>G, EGFR L858R, and NM_007294.4(BRCA1):c.5558dup (p.Tyr1853Ter). 
- DefiningLocationConstraint: GRCh38/hg38 7p22.1(chr7:5905831-6014161)x3, GRCh38 Xp22.31(chrX:6978350-7594949)x3, and BRAF V600
- CopyCountConstraint: 3 copies
- CopyChangeConstraint: Gain and Loss
- FeatureContextConstraint: TP53

Examples shown for Recipes display the entire example. Inline examples highlighted are:
- CanonicalAllele: NM_004958.4(MTOR):c.5992_5993del (p.Met1998fs) and NC_000001.11:g.1699974C>G
- ProteinSequenceConsequence: EGFR L858R and NM_007294.4(BRCA1):c.5558dup (p.Tyr1853Ter)
- CategoricalCnv: GRCh38/hg38 7p22.1(chr7:5905831-6014161)x3 with CopyCountConstraint, GRCh38/hg38 7p22.1(chr7:5905831-6014161)x3 with CopyChangeConstraint, GRCh38 Xp22.31(chrX:6978350-7594949)x3, and TP53 Loss.

I locally ran tests and pre-commit hooks on the branch before opening this Pull Request.

Pull request check list:
- [x] Does the title of this Pull Request [reference the corresponding Issue](https://github.com/ga4gh/cat-vrs/blob/1.x/CONTRIBUTING.md#pull-requests)?
- [x] Is the branch validating against pre-commit hooks? Run `pre-commit run --all-files` from the root directory.
- [x] Is the branch passing tests? Run `pytest tests/` from the root directory.

As an aside, examples/ are currently named based on the Recipe that they satisfy and at the moment the BRAF V600 example does not satisfy any Recipe. It uses DefiningLocation and FeatureContext constraints to generally highlight the 600th amino acid within BRAF. We may want to revise example naming schemes in the future but we'll see.

brendanreardon committed:
In progress BRAF V600 and categorical cnv with ranges

DanielPuthawala committed:
addded TP-53 CNLoss variant example

DanielPuthawala committed:
fixing trailing whitespace

DanielPuthawala committed:
fixing trailing whitespace 2

brendanreardon committed:
Updated comments and documentation with ex3

cmprocknow committed:
added a member

sabrinacamp2 committed:
Full examples of BRAF V600 members

sabrinacamp2 committed:
Merge remote-tracking branch 'origin/87-even-more-examples' into 87-even-more-examples

brendanreardon committed:
Removed extra line

DanielPuthawala committed:
fix trailing whitespace

brendanreardon committed:
Added spaces

DanielPuthawala committed:
remove whitespace

brendanreardon committed:
#133: Import examples into readthedocs (#143)

* added ccnv-ex3 to examples-source.yaml

* Examples of using collapse, including yaml, and highlighting lines

* Added sphinx extension for collapse

* added maturity status and description to catvar example

* added canonicalAllele-ex1

* added canonicalAllele-ex2

* prepended 'example_' to examples

* added example_categoricalCNV-ex1

* oops, -ex2

* added real example_categoricalCNV-ex1

* example_describedVariant-ex1

* example_proteinMarker-ex1

* example_proteinSequenceConsequence-ex1

* example_proteinSequenceConsequence-ex2

* Added sphinx toolbox to .requirements.txt

* Moved toolbox requirements to requirements within docs folder

* Readded empty line

* Create README.md

created README for adding exampels to the examples-source.yaml file

* Fixed typos

---------

Co-authored-by: Daniel Puthawala <140105309+DanielPuthawala@users.noreply.github.com>
Co-authored-by: DanielPuthawala <Daniel.Puthawala@nationwidechildrens.org>

brendanreardon committed:
Expanded upon sequence location and reference for canonical allele example 1

brendanreardon committed:
Added more inforamtion for location

brendanreardon committed:
More details for canonicalAllele-ex1

brendanreardon committed:
Expanded upon details within a few examples

brendanreardon committed:
Added information to location and sequence reference for categoricalcnv-ex1 and -ex2

brendanreardon committed:
variant normalizer unable to tokenize the grch37 equivalent to add a member

brendanreardon committed:
Revised catcnv1, catcnv2

brendanreardon committed:
Updated categoricalcnv-ex4

brendanreardon committed:
Fixed error with categoricalcnv-ex1

brendanreardon committed:
Fixed trailing whitespace

brendanreardon committed:
removed name from location

brendanreardon committed:
Updated proteinsequenceconsequence-ex1.yaml

brendanreardon committed:
Updated protein sequence consequence examples

brendanreardon committed:
Updated README

brendanreardon committed:
Updated README and annotated protein sequence consequence 3

brendanreardon committed:
Compiled def/ and json/ version of examples

brendanreardon committed:
json for protein sequence consequence ex1

brendanreardon committed:
Updated examples for recipes

brendanreardon committed:
Added drop down menu examples to constraints and recipes

brendanreardon committed:
Fixed end of line skip

brendanreardon committed:
Removed BRAF V600 from protein sequence consequence Recipe inline example

brendanreardon commented:
A bolded "Examples" header were added to each concept on readthedocs with examples listed under. For example, here is DefiningAlleleConstraint:

<img width="607" alt="DefiningAlleleConstraint in readthedocs with examples" src="https://github.com/user-attachments/assets/d3cc9bc9-b614-4325-b82b-e84c2c22d0b0" />

***

All examples are collapsed by default. Clicking on any example will expand it and show the example in JSON format, such as this screenshot for CopyChangeConstraint:

<img width="617" alt="CopyChangeConstraint in readthedocs with examples" src="https://github.com/user-attachments/assets/6391e882-2f80-44d3-b784-dd5b0b5fad47" />

***

As written in the Pull Request, Recipes will show the full example file. For example, here is a snippet of an example for CanonicalAllele:

<img width="569" alt="CanonicalAllele in readthedocs with examples" src="https://github.com/user-attachments/assets/be301daf-fbae-4d86-9f22-90b54cfeb418" />

DanielPuthawala reviewed (approved):
Really strong work here!  

I see nothing that prevents us from going ahead with release.

ahwagner commented on examples/braf-v600.yaml:
```diff
@@ -0,0 +1,291 @@
+    description: VRS variation of NM_004333.6:c.1799T>A, generated with the VICC Variation Normalizer (https://github.com/cancervariants/variation-normalization).
+    digest: W6xsV-aFm9yT2Bic5cFAV2j0rll6KK5R
+    expression:
+      - syntax: hgvs.g
+        value: NC_000007.14:g.140753336A>T
+      - syntax: hgvs.c
+        value: NM_004333.6:c.1799T>A
+      - syntax: hgvs.p
+        value: NP_004324.2:p.Val600Glu
```
I'm not a huge fan of using all 3 HGVS syntax forms here. The Allele is only the `c.` variant.

ahwagner replied:
This might be more of a comment for @korikuzma to revise the VICC variation normalizer–but we should drop it from this example either way.

ahwagner replied:
```suggestion
      - syntax: hgvs.c
        value: NM_004333.6:c.1799T>A
```

ahwagner commented on examples/braf-v600.yaml:
```diff
@@ -0,0 +1,291 @@
+      - syntax: hgvs.c
+        value: NM_004333.6:c.1798_1799delinsAA
+      - syntax: hgvs.p
+        value: NP_004324.2:p.Val600Lys
```
```suggestion
      - syntax: hgvs.c
        value: NM_004333.6:c.1798_1799delinsAA
```

ahwagner commented on examples/braf-v600.yaml:
```diff
@@ -0,0 +1,291 @@
+      - syntax: hgvs.c
+        value: NM_004333.6:c.1799T>G
+      - syntax: hgvs.p
+        value: NP_004324.2:p.Val600Gly
```
```suggestion
      - syntax: hgvs.c
        value: NM_004333.6:c.1799T>G
```

ahwagner commented on examples/braf-v600.yaml:
```diff
@@ -0,0 +1,291 @@
+      - syntax: hgvs.c
+        value: NM_004333.6:c.1798G>A
+      - syntax: hgvs.p
+        value: NP_004324.2:p.Val600Met
```
```suggestion
      - syntax: hgvs.c
        value: NM_004333.6:c.1798G>A
```

ahwagner commented on examples/braf-v600.yaml:
```diff
@@ -0,0 +1,291 @@
+      - syntax: hgvs.c
+        value: NM_004333.6:c.1798_1799delinsAG
+      - syntax: hgvs.p
+        value: NP_004324.2:p.Val600Arg
```
```suggestion
      - syntax: hgvs.c
        value: NM_004333.6:c.1798_1799delinsAG
```

ahwagner commented on docs/source/concepts/catvrs_model.rst:
```diff
@@ -34,28 +34,158 @@ DefiningAlleleConstraint
+      :language: json
+      :lines: 83-172
+
+.. raw:: html
+
+   <div style="margin-top: 0.5em;"></div>
```
Raw HTML repeated regularly like this is a sign we need to update our .css. This should be handled as a separate issue, created #149 to address later.

brendanreardon replied:
Sounds good!

ahwagner reviewed (approved):
This is a huge PR, with over 5000 added lines. In general this looks pretty good, but I may have missed some things given the size. Recommend merging then reassessing final product for any observed issues.

brendanreardon committed:
Update examples/braf-v600.yaml

Good to know, thank you! I'll take it that we should _only_ include the expressions for the subject Allele itself. Larger question - is it the intent / kosher for related Alleles that translate to an amino acid that satisfy the subject catvar be included under the `members` property? e.g., hgvs.g Alleles that translate to a BRAF V600 amino acid change.

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

brendanreardon committed:
Update examples/braf-v600.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

brendanreardon committed:
Update examples/braf-v600.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

brendanreardon committed:
Update examples/braf-v600.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

brendanreardon committed:
Update examples/braf-v600.yaml

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

brendanreardon commented:
@ahwagner thank you for the suggestions on limiting what we include in expressions! Thank you! I'll take it that we should _only_ include the expressions for the subject Allele itself. 

Double checking - it _is_ the intent of the `members` field though to include related Alleles, right? For example, we could update the BRAF V600 example to include the VRS Allele for "NC_000007.14:g.140753336A>C" as a member. My understanding is that this is okay because the `relations` field of DefiningAlleleConstraint and DefiningLocationConstraint include translation_of codings.

Edit: We also included genomic positions as extensions for location objects for hgvs.c Alleles per [Issue 144](https://github.com/ga4gh/cat-vrs/issues/144). Would you likewise recommend only including genomic positions for hgvs.g Alleles?  

For example, 👇 
<img width="1027" alt="Screenshot 2025-06-10 at 11 16 18 AM" src="https://github.com/user-attachments/assets/ea643fe5-a5ad-48ec-8e4b-c551eadf2977" />

brendanreardon committed:
Updated BRAF V600 example in annotated example, examples source, and json version per Alex recommending removing hgvs.g and hgvs.p expressions from members
